client/src/components/blocks/TopTimeBox/index.tsx

```diff
@@ -1,18 +1,23 @@
-import { useRef } from "react";
+import { useEffect, useState } from "react";
 import TopTimeBar from "components/atoms/TopTimeBar";
 import "./style.scss";
 
-function TopTimeBox({ bpm = 0 }) {
-  const topTimeBoxRef = useRef<HTMLDivElement>(null);
-  const stickCnt = Math.floor(bpm / 2) + 1;
+function TopTimeBox({ bpm = 30 }) {
+  const [stickCnt, setStickCnt] = useState(bpm);
+  const [timeBoxSize, setTimeBoxSize] = useState(
+    (document.querySelector("#track-space") as HTMLDivElement)?.offsetWidth,
+  );
 
   const gapBetweenBars = (isLast = false) => {
+    console.log("between");
     const WIDTH_OF_TIME_BAR = 4;
     const PADDING_LEFT = 4;
     const WIDTH_OF_PLAY_STICK = 12;
-    if (!topTimeBoxRef.current) return 0;
+
+    const trackSpaceDOM = document.querySelector("#track-space");
+    if (!trackSpaceDOM) return 0;
     const timeBoxWidth =
-      (document.querySelector("#track-space") as HTMLDivElement).offsetWidth -
+      (trackSpaceDOM as HTMLDivElement).offsetWidth -
       parseFloat(getComputedStyle(document.querySelector("#track-space") as HTMLDivElement).paddingLeft) -
```

Comment:
변수로 선언했으니 기왕이면 이 친구도 바꿔주세요.. 😀

```suggestion
      parseFloat(getComputedStyle(trackSpaceDOM as HTMLDivElement).paddingLeft) -
```

Reply:
반영했습니다! 확인 부탁드립니다ㅎㅎ